### PR TITLE
New package: headscale-0.16.4

### PR DESCRIPTION
**Pull request by teldra**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

### CODE REVIEW

**Comment by teldra** [comment]
Crosscompiling fails, because it compiles for host? How do I circumvent this?

**Comment by paper42** [comment]
use the go build style

**Comment by teldra** [comment]
> use the go build style

I tried but I don't get a binary. At least I don't find any

**Comment by paper42** [comment]
I tried to package headscale, but didn't end up finishing it, feel free to get inspiration from here: 29f7be7ad2a8b362b584791c7cdbdf82c62e19b9

**Comment by teldra** [comment]
Thank you @paper42, that did the trick!

File: srcpkgs/headscale/template [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+maintainer="teldra <teldra@rotce.de>"
+license="BSD-3-Clause"
+homepage="https://github.com/juanfont/headscale"
+changelog="https://raw.githubusercontent.com/juanfont/headscale/main/CHANGELOG.md"
+distfiles="https://github.com/juanfont/headscale/archive/refs/tags/v${version}.tar.gz"
+checksum=0395478f9dde68aa8ca23be8df6ff636d47166981d0995e4e31a8c7db12df8e8
+make_dirs="/etc/headscale 0750 _headscale _headscale
+	/var/lib/headscale 0750 _headscale _headscale
+	/var/run/headscale 0750 _headscale _headscale"
+system_accounts="_headscale _headscale"
```
[the remark this answers is not in the document]
Comment (teldra) [reply srcpkgs/headscale/template:19]: Removed redundant word.

File: srcpkgs/headscale/template [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+go_ldflags="-X ${go_import_path}/cmd/headscale/cli.Version=${version}"
+short_desc="Self-hosted implementation of the Tailscale control server"
+maintainer="teldra <teldra@rotce.de>"
+license="BSD-3-Clause"
+homepage="https://github.com/juanfont/headscale"
+changelog="https://raw.githubusercontent.com/juanfont/headscale/main/CHANGELOG.md"
+distfiles="https://github.com/juanfont/headscale/archive/refs/tags/v${version}.tar.gz"
+checksum=0395478f9dde68aa8ca23be8df6ff636d47166981d0995e4e31a8c7db12df8e8
+make_dirs="/etc/headscale 0750 _headscale _headscale
+	/var/lib/headscale 0750 _headscale _headscale
```
[the remark this answers is not in the document]
Comment (teldra) [reply srcpkgs/headscale/template:17]: I removed the line.

File: srcpkgs/headscale/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	/var/run/headscale 0750 _headscale _headscale"
```
[the remark this answers is not in the document]
Comment (teldra) [reply srcpkgs/headscale/template:18]: `headscale` creates a sock file in this dir. (Please refer to the vsed statement in `post_extract` and maybe the config.yaml.)
I think, headscale is originally not intended to run as unprivileged user. I circumvent this with the creation of a headscale owned folder.

[the remark this answers is not in the document]
Comment (teldra) [reply srcpkgs/headscale/template:18]: I try to maximize the ootb experience.

File: srcpkgs/headscale/template [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+go_package="cmd/headscale/headscale.go"
+go_ldflags="-X ${go_import_path}/cmd/headscale/cli.Version=${version}"
+short_desc="Self-hosted implementation of the Tailscale control server"
+maintainer="teldra <teldra@rotce.de>"
+license="BSD-3-Clause"
+homepage="https://github.com/juanfont/headscale"
+changelog="https://raw.githubusercontent.com/juanfont/headscale/main/CHANGELOG.md"
+distfiles="https://github.com/juanfont/headscale/archive/refs/tags/v${version}.tar.gz"
+checksum=0395478f9dde68aa8ca23be8df6ff636d47166981d0995e4e31a8c7db12df8e8
+make_dirs="/etc/headscale 0750 _headscale _headscale
```
[the remark this answers is not in the document]
Comment (teldra) [reply srcpkgs/headscale/template:16]: Added `vinstall` of the conf file.

File: srcpkgs/headscale/template [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+short_desc="Self-hosted implementation of the Tailscale control server"
+maintainer="teldra <teldra@rotce.de>"
+license="BSD-3-Clause"
+homepage="https://github.com/juanfont/headscale"
+changelog="https://raw.githubusercontent.com/juanfont/headscale/main/CHANGELOG.md"
+distfiles="https://github.com/juanfont/headscale/archive/refs/tags/v${version}.tar.gz"
+checksum=0395478f9dde68aa8ca23be8df6ff636d47166981d0995e4e31a8c7db12df8e8
+make_dirs="/var/run/headscale 0770 _headscale _headscale"
+system_accounts="_headscale"
+_headscale_homedir="/var/lib/headscale"
```
Comment (teldra) [inline srcpkgs/headscale/template:18]: Anyhow, this is not created.

File: srcpkgs/headscale/template (hunk 2 above)
[the remark this answers is not in the document]
Comment (teldra) [reply srcpkgs/headscale/template]: This folder is not created with the `system_accounts` feature.

[the remark this answers is not in the document]
Comment (0x5c) [reply srcpkgs/headscale/template]: Sorry about that, I misread the flags passed to useradd in the trigger and their help text, you indeed have to create the homedir manually in make_dirs.

File: srcpkgs/headscale/template (hunk 3 above)
[the remark this answers is not in the document]
Comment (0x5c) [reply srcpkgs/headscale/template]: Ok yeah that dir is needed.

I think systemd (which they kinda design for) makes `/var/run` writable.

[the remark this answers is not in the document]
Comment (0x5c) [reply srcpkgs/headscale/template]: Just realised that `/var/run` is a symlink to `/run`, a tmpfs; the directory will not survive a reboot.

The [manual](https://github.com/void-linux/void-packages/blob/master/Manual.md#writing-runit-services) says to create the dir in the service file.

**Pull request closed without merging**